cscs-checks/analytics/hpcg/hpcg_benchmark.py

```diff
@@ -0,0 +1,109 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+@rfm.simple_test
+class HPCGCheckRef(rfm.RegressionTest):
+    def __init__(self, **kwargs):
+        super().__init__()
+
+        self.descr = 'HPCG reference benchmark'
+        self.valid_systems = ['daint:mc', 'daint:gpu', 'dom:gpu', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.modules = ['craype-hugepages8M']
+        self.build_system = 'Make'
+        self.build_system.options = ['arch=MPI_GCC_OMP']
+        self.sourcesdir = None
+        self.prebuild_cmd = ['git clone https://github.com/hpcg-benchmark/hpcg.git', 'cd hpcg']
+
+        self.executable = 'hpcg/bin/xhpcg'
+        self.executable_opts = ['--nx=104', '--ny=104', '--nz=104', '-t2']
+        output_file = sn.getitem(sn.glob('HPCG*.txt'), 0)
+        self.sanity_patterns = sn.assert_eq(4, sn.count(
+            sn.findall(r'PASSED', output_file)))
+        self.num_tasks = 12
```

Comment:
Should the number of tasks depend on the system:partition combo? I think we should be using a different number for daint/dom:mc.

Reply:
not that it makes a difference for performance in this case, but for completeness I added this